datafusion-jni/src/stream.rs

```diff
@@ -0,0 +1,110 @@
+use arrow::array::Array;
+use arrow::array::StructArray;
+use arrow::ffi::{FFI_ArrowArray, FFI_ArrowSchema};
+use datafusion::physical_plan::SendableRecordBatchStream;
+use futures::stream::TryStreamExt;
+use jni::objects::{JClass, JObject, JValue};
+use jni::sys::jlong;
+use jni::JNIEnv;
+use std::convert::Into;
+use std::ptr::addr_of_mut;
+use tokio::runtime::Runtime;
+
+#[no_mangle]
+pub extern "system" fn Java_org_apache_arrow_datafusion_DefaultRecordBatchStream_next(
+    env: JNIEnv,
```

Comment:
need to use mut: https://github.com/jni-rs/jni-rs/blob/master/docs/0.21-MIGRATION.md